scipy/interpolate/fitpack2.py

```diff
@@ -1934,8 +1934,8 @@ def __init__(self, u, v, r, s=0., pole_continuity=False, pole_values=None,
         u, v = np.ravel(u), np.ravel(v)
         r = np.asarray(r)
 
-        if not ((0.0 <= u).all() and (u <= np.pi).all()):
-            raise ValueError('u should be between [0, pi]')
+        if not (0.0 < u[0] and u[-1] < np.pi):
```

Comment:
just to note: dropping `all()` here is fine, this is a little more efficient and moves the rest of the check to the monotonicity validation further down.